### PR TITLE
Feedback on updating API landing page

### PR DESCRIPTION
**Pull request by katrinabrock**
I'd like to make the [API landing page](http://docs.diesel.rs/diesel/index.html) a little more useful/welcoming. This is my first incremental step. Looking for feedback not just on this, but on what else should go here. Perhaps a debugging section including the info in #290? API Overview? (What would that entail?) I've talked to @sgrif a bit about this an d am interested in getting feedback from others.

### CODE REVIEW

File: diesel/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,17 @@
...
-//! found on our website.
+//!# Diesel
+//!
+//! Diesel is an ORM and query builder designed to reduce the boilerplate for database interactions. 
+//!
+//! * [Homepage](http://diesel.rs/)
+//! * [Getting Started Guide](http://diesel.rs/guides/getting-started/)
+//! * [Source Code](https://github.com/diesel-rs/diesel/)
+//!
+//!## Crates
```
Comment (katrinabrock) [inline diesel/src/lib.rs]: Not sure this is the best heading, but this list needs a heading. Suggestions?

Comment (juliusdelta) [reply diesel/src/lib.rs]: [Rocket uses 'Libraries'](https://api.rocket.rs/rocket/) which I think is what I would suggest here. Super stoked for this to be done btw.

**Comment by juliusdelta** [comment]
Maybe a configuration section would be good with different config options and supported DB's? Rocket has a brief one on their [docs](https://api.rocket.rs/rocket/#configuration).

Like:
```
diesel = { version = "0.15.0", features = ["postgres"] }
diesel_codegen = { version = "0.15.0", features = ["postgres"] }
```

File: diesel/src/lib.rs (hunk 1 above)
Comment (katrinabrock) [reply diesel/src/lib.rs]: "Libraries" sounds nicer, but is a CLI tool really a library?

Comment (juliusdelta) [reply diesel/src/lib.rs]: Good point. Hm. I'd almost suggest `Crates & Libraries` but there's only 3 things.

File: diesel/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,17 @@
-//! Diesel is an ORM and query builder designed to reduce the boilerplate for database
-//! interactions. [A getting started guide](http://diesel.rs/guides/getting-started/) can be
-//! found on our website.
+//!# Diesel
```
Comment (killercup) [inline diesel/src/lib.rs]: Can you add a space after the `//!` and before the `#`? (Same below)

File: diesel/src/lib.rs (hunk 1 above)
Comment (killercup) [reply diesel/src/lib.rs]: So I spend a surprisingly long amount of time reading the wikipedia pages for gasoline and diesel to see what the things that make up the fuel are called… and all I found was "component"

Comment (notryanb) [reply diesel/src/lib.rs]: `Tools & Libraries`? 🤔 I think what you have written does a good job (Crates as the heading). Technically, each one of those "components" is a crate and there is a short description detailing its purpose. I don't see any ambiguous language that would confuse someone or inhibit them from getting to the correct documentation from your links. 

... although, they are "containers" that "hold diesel" code. We can call them `Tanks`.. 😆

**Comment by katrinabrock** [comment]
fixed spacing :-)

**Comment by killercup** [comment]
[Very close!](https://travis-ci.org/diesel-rs/diesel/jobs/282150517) I'll make rustfmt happy and merge this :)

**Comment by katrinabrock** [comment]
Merp. I need to start running that locally, huh? Thanks @killercup.

**Comment by killercup** [comment]
No worries :) (I set you editor to strip trailing whitespace in .rs files)

**Pull request merged**